src/main/java/com/aws/greengrass/device/session/MqttSessionFactory.java

```diff
@@ -31,14 +33,22 @@ public Session createSession(Map<String, String> credentialMap) throws Authentic
         //   we don't yet have the ability to authorize actions for those clients. So, for now, assume
         //   this is an IoT Thing and components will be specially handled elsewhere.
         Thing thing = new Thing(mqttCredential.clientId);
-        Certificate cert = new Certificate(mqttCredential.certificatePem);
-        if (!iotAuthClient.isThingAttachedToCertificate(thing, cert)) {
-            throw new AuthenticationException("unable to authenticate device");
+        Optional<String> certificateId;
+        try {
+            certificateId = iotAuthClient.getActiveCertificateId(mqttCredential.certificatePem);
```

Comment:
@jbutler.

By this point we've already done this same call once: https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/0a726d2c354c4cefd8e827f40caf2b8cf8d3fcd1/src/main/java/com/aws/greengrass/ipc/VerifyClientDeviceIdentityOperationHandler.java#L109. 

We'd prefer not to do it again right. How do you suggest we make that work?

Reply:
This is fine for now. We'll need a caching layer for certs and things, which we can add later.